pyproject.toml

```diff
@@ -153,28 +153,30 @@ skip-string-normalization = 1
 
 [tool.ruff]
 line-length = 88
-target-version = 'py310'
+exclude = [
+    '.git',
+    '.ruff_cache',
+    'build',
+    'build-install',
+    'dist',
+    'doc/source/auto_examples',
+]
```

Comment:
Why is this necessary? Since we only use ruff via pre-commit hooks I thought, that we can rely on `.gitignore` to cover this?

Reply:
It was added in #6729. I am not sure why. 

According to https://docs.astral.sh/ruff/settings/#exclude, `.git` and `.ruff_cache` are ignored by default and `build`, `build-install`, `dist`, and `doc/source/auto_examples/**/*` are in `.gitignore`.

I can try deleting and see what happens.

Reply:
Ah, my bad. I didn't notice that the exclude-block was just moved. I thought this PR added the block as new configuration and couldn't explain it. Sorry for the confusion. I'm happy to go with either way that works. Though, relying on `.gitignore` should be more robust and less repetition.

Reply:
And now `pyproject.toml` is 9 lines shorter!